### ISSUE TITLE
call-datetime-strptime-without-zone (DTZ007) false positive with safe method chain

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ruff#28014]
### Summary

```python
from datetime import datetime

datetime.now().replace(microsecond=0).astimezone()                  # Accepted
datetime.strptime("2026", "%Y").replace(microsecond=0).astimezone() # DTZ007
```

Both of these produce aware datetimes, but DTZ007 still alerts on the second line. An analogous bug was fixed in #15998.

### Version

0.16.4
